roles/kubernetes/client/tasks/main.yml

```diff
@@ -3,7 +3,7 @@
   set_fact:
     external_apiserver_address: >-
       {%- if loadbalancer_apiserver is defined and loadbalancer_apiserver.port is defined -%}
-      {{ apiserver_loadbalancer_domain_name }}
+      {{ loadbalancer_apiserver.address }}
```

Comment:
AWS ELB doesn't give a fixed ip, but the domain name is consistent. If anything, we should point to domain name first, then address if it's not defined.

Reply:
I understand that but `external_apiserver_address` is getting used here:

```
- name: Generate admin kubeconfig with external api endpoint
  shell: >-
    {{ bin_dir }}/kubeadm alpha
    {% if kubeadm_version is version('v1.13.0', '<') %}
    phase
    {% endif %}
    kubeconfig user
    --client-name kubernetes-admin
    --org system:masters
    --cert-dir {{ kube_config_dir }}/ssl
    --apiserver-advertise-address {{ external_apiserver_address }}
    --apiserver-bind-port {{ external_apiserver_port }}
```

When you use `apiserver_loadbalancer_domain_name` as the `external_apiserver_address` here, this command will fail:

```
# kubeadm alpha kubeconfig user --client-name kubeadmin-test --org system:masters --cert-dir /etc/kubernetes/ssl --apiserver-advertise-address vip-master.example.com --apiserver-bind-port 443
couldn't use "vip-master.example.com" as "apiserver-advertise-address", must be ipv4 or ipv6 address
```

There is no way currently to provide a domain name to `kubeadm alpha kubeconfig user` command:

```
  kubeadm alpha kubeconfig user --client-name=foo

Flags:
      --apiserver-advertise-address string   The IP address the API server is accessible on
      --apiserver-bind-port int32            The port the API server is accessible on (default 6443)
      --cert-dir string                      The path where certificates are stored (default "/etc/kubernetes/pki")
      --client-name string                   The name of user. It will be used as the CN if client certificates are created
  -h, --help                                 help for user
      --org strings                          The orgnizations of the client certificate. It will be used as the O if client certificates are created
      --token string                         The token that should be used as the authentication mechanism for this kubeconfig, instead of client certificates

Global Flags:
      --log-file string   If non-empty, use this log file
      --rootfs string     [EXPERIMENTAL] The path to the 'real' host root filesystem.
      --skip-headers      If true, avoid header prefixes in the log messages
  -v, --v Level           log level for V logs
```

Reply:
@EppO Thanks for the feedback. I will raise this issue with the kubeadm devels.